### PR TITLE
Introduce algolia/search-bundle v3

### PR DESCRIPTION
**Pull request by julienbourdeau**
| Q             | A
| ------------- | ---
| License       | MIT

There is already a recipe for the Algolia Search Bundle and I have just released the version 3.0. Version 2 will be maintained (patch-only) but should not be installed for new projects.

This version introduce many new features but mostly:

* **Simple**: You can get started with only 5 lines of YAML
* **Extensible**: It let’s you easily replace services by implementing Interfaces
* **Standard**: It leverages `Normalizers` to convert entities for indexing
* **Dev-friendly**: It lets you disable HTTP calls easily (while running tests for examples)
* **Futur-ready**: It let’s you unsubscribe from doctrine events easily to use a messaging/queue system.

Documentation can be found [here](http://www.algolia.com/doc/api-client/symfony/) or in the [README](https://github.com/algolia/search-bundle/blob/master/README.md).

Any feedback is very welcomed 😃

### CODE REVIEW

File: algolia/algolia-search-bundle/2.2/config/packages/algolia.yaml [hunk 1]
```diff
@@ -1,3 +0,0 @@
-algolia:
```
Comment (Nyholm) [inline algolia/algolia-search-bundle/2.2/config/packages/algolia.yaml]: No need to remove the old recipe

Comment (julienbourdeau) [reply algolia/algolia-search-bundle/2.2/config/packages/algolia.yaml]: I think it's important to remove it, people should use it anymore. I believe this is why the tests are failing. I'm not sure how to fix this 🤔

Comment (Nyholm) [reply algolia/algolia-search-bundle/2.2/config/packages/algolia.yaml]: If I really want to install 2.2 for some reason, shouldn't I be able to use the old recipe?

**Comment by julienbourdeau** [comment]
In the end, I have restored the recipe for 2.2. We can remove later if it makes sense, you're right @Nyholm it might be too early now. 👍

**Comment by julienbourdeau** [comment]
Thanks @Nyholm 👌

**Pull request merged**